migration-engine/connectors/sql-migration-connector/src/lib.rs

```diff
@@ -151,6 +151,26 @@ impl SqlMigrationConnector {
         Ok(())
     }
 
+    /// For tests.
+    pub fn migration_from_schemas(
+        from: (&Configuration, &Datamodel),
+        to: (&Configuration, &Datamodel),
+    ) -> SqlMigration {
+        let connection_info = ConnectionInfo::from_url(&from.0.datasources[0].load_url().unwrap()).unwrap();
+        let flavour = flavour::from_connection_info(&connection_info);
+        let from_sql = sql_schema_calculator::calculate_sql_schema(from, flavour.as_ref());
+        let to_sql = sql_schema_calculator::calculate_sql_schema(to, flavour.as_ref());
+
+        let steps = sql_schema_differ::calculate_steps(Pair::new(&from_sql, &to_sql), flavour.as_ref());
+
+        SqlMigration {
+            before: from_sql,
+            after: to_sql,
+            added_columns_with_virtual_defaults: Vec::new(),
+            steps,
+        }
+    }
```

Comment:
Ideally, we will have a version of `MigrationConnector::diff()` that doesn't take `self` in the future.